### ISSUE TITLE
Sound testing error

### ISSUE THREAD

**Issue by webmarco78** [bablokb/pi-btaudio#9]
I am following tutorial for the [sPot](https://github.com/dupontgu/retro-ipod-spotify-client) project.
I installed pi-btaudio and configured it by following your guide.
A BT Speaker is connected and when I use: aplay -f cd /usr/share/sounds/alsa/*
I hear first file (Front_Center) then audio stops and the messages on console are:
Playing WAVE '/usr/share/sounds/alsa/Front_Center.wav' : Signed 16 bit Little Endian, Rate 48000 Hz, Mono
Playing WAVE '/usr/share/sounds/alsa/Front_Left.wav' : Signed 16 bit Little Endian, Rate 48000 Hz, Mono
aplay: set_params:1407: Unable to install hw params:
ACCESS:  RW_INTERLEAVED
FORMAT:  S16_LE
SUBFORMAT:  STD
SAMPLE_BITS: 16
FRAME_BITS: 16
CHANNELS: 1
RATE: 48000
PERIOD_TIME: 125000
PERIOD_SIZE: 6000
PERIOD_BYTES: 12000
PERIODS: 4
BUFFER_TIME: 500000
BUFFER_SIZE: 24000
BUFFER_BYTES: 48000
TICK_TIME: 0

I would like testing audio via Bluetooth before going on with Spotify setup and other stuff in the project.
Any suggestion about this issue?

**Comment by bablokb** [comment]
Have you tried playing one file at a time? It would be interesting if this behaves differently. When I google your error-message, it seems this might not be related to BT at all.

**Comment by webmarco78** [comment]
If I play only one file at time, it souds properly via the BT speaker.
Googling around for this issue, I read something about the single-user access needed by aplay or something related to user groups.
I am still investigating.
Did you have same behaviour on your system?

**Comment by bablokb** [comment]
No, I don't think it has to do with permissions. The "Unable to install hw params" error also seems to pop up in other contexts, not only BT. It might have to do with stack-corruption. Playing one file at a time will reset the stack, so it works (but you probably can't rely on it).

BTW: besides aplay I always install mpg123, a commandline mp3-player and very useful for testing.

Did you build bluez-alsa from source? If not, this might be a way to go. There was an issue in the source relating to your problem, and it might be fixed in a newer version.

**Comment by webmarco78** [comment]
I didn't install bluez-alsa from source, I'll try it even if I am not that expert in building from sources.
Furthermore I'll test mpg123 and test it in play.
Thank you for your support

**Comment by webmarco78** [comment]
I started from scratch and now the bluetooth is working.
The only "problem"  is that the speaker doesn't recconnect after reboot of rpi. I need to execute btaudio-connect from command line (via ssh since it is a headless system)

**Comment by bablokb** [comment]
Yes, that is documented in the Readme (at least for Bullseye 2021-11). Which  version did you use?
